java/arcs/core/data/proto/TypeProtoDecoders.kt

```diff
@@ -82,6 +83,12 @@ fun CountTypeProto.decode() = CountType()
 /** Converts a [TupleTypeProto] protobuf instance into a Kotlin [TupleType] instance. */
 fun TupleTypeProto.decode() = TupleType(elementsList.map { it.decode() })
 
+/** Converts a [TypeVariableProto] protobuf instance into a Kotlin [TypeVariable] instance. */
+fun TypeVariableProto.decode() = TypeVariable(
+    name,
+    if (hasConstraint()) constraint.constraintType.decode() else null
```

Comment:
isn't that: `constraint?.constraintType?.decode()` ?

Reply:
Nice, will update.

Reply:
Strangely, this fails the unit test.

Reply:
Unfortunately, proto fields are not null if they are not set. So you have you explicitly check.